sklearn/semi_supervised/label_propagation.py

```diff
@@ -449,6 +490,10 @@ class LabelSpreading(BaseLabelPropagation):
     def __init__(self, kernel='rbf', gamma=20, n_neighbors=7, alpha=0.2,
                  max_iter=30, tol=1e-3, n_jobs=1):
 
+        if alpha <= 0.0 or alpha >= 1.0:
```

Comment:
In `fit` because of convention?

Otherwise, I would argue that failing as early as possible (i.e. in `__init__`) would be better.